src/benchmark_llm_serving/backends.py

```diff
@@ -0,0 +1,226 @@
+
+        Returns:
+            dict : The useful metrics
+        """
+        metrics = {}
+        metrics['num_requests_running'] = metrics_dict['vllm:num_requests_running'][0]['value']
+        metrics['num_requests_waiting'] = metrics_dict['vllm:num_requests_waiting'][0]['value']
+        metrics['gpu_cache_usage_perc'] = metrics_dict['vllm:gpu_cache_usage_perc'][0]['value']
+        return metrics
+
+
+class BackEndMistral(BackEnd):
+
+    def get_payload(self, query_input: QueryInput, args: argparse.Namespace) -> dict:
+        """Gets the payload to give to the model
+
+        Args:
+            query_input (QueryInput) : The query input to use
+            args (argparse.Namespace) : The cli args
+
+        Returns:
+            dict : The payload
+        """
+        return {"messages": [{"role": "user", "content": query_input.prompt}],
+                        "model": args.model,
+                        "max_tokens": args.output_length,
+                        "min_tokens": args.output_length,
+                        "temperature": self.TEMPERATURE,
+                        "stream": True
+                                }
+        
+    def test_chunk_validity(self, chunk: str) -> bool:
+        """Tests if the chunk is valid or should not be considered.
+
+        Args:
+            chunk (str) : The chunk to consider
+
+        Returns:
+            bool : Whether the chunk is valid or not
+        """
+        if chunk[:4] == "tok-":
+            return False
+        else:
+            return True
+
+    def get_completions_headers(self) -> dict:
+        """Gets the headers (depending on the backend) to use for the request
+
+        Returns:
+            dict: The headers
+
+        """
+        return {"Accept": "application/json",
+                "Content-Type": "application/json"}
+
+    def get_newly_generated_text(self, json_chunk: dict) -> str:
+        """Gets the newly generated text
+
+        Args:
+            json_chunk (dict) : The chunk containing the generated text
+
+        Returns:
+            str : The newly generated text
+        """
+        if len(json_chunk['choices']):
+            data = json_chunk['choices'][0]['delta']["content"]
+            return data
+        else:
+            return ""
+
+
+def get_backend(backend_name: str) -> BackEnd:
+    implemented_backends = ["mistral", "happy_vllm"]
+    if backend_name not in implemented_backends:
+        raise ValueError(f"The specified backend {backend_name} is not implemented. Please use one of the following : {implemented_backends}")
+    if backend_name == "happy_vllm":
+        return BackendHappyVllm(backend_name, chunk_prefix="data: ", last_chunk="[DONE]", metrics_endpoint_exists=True)
+    if backend_name == "mistral":
+        return BackEndMistral(backend_name, chunk_prefix="data: ", last_chunk="[DONE]", metrics_endpoint_exists=False)
+    return BackEnd("not_implemented")
```

Comment:
Not sure, but is there a scenario that reaches this stage ? `return BackEnd("not_implemented")`